library/src/scala/compiletime/package.scala

```diff
@@ -168,13 +164,8 @@ transparent inline def summonInline[T]: T =
  *  @return the given values typed as elements of the tuple
  */
 inline def summonAll[T <: Tuple]: T =
-  val res =
-    inline erasedValue[T] match
-      case _: EmptyTuple => EmptyTuple
-      case _: (t *: ts) => summonInline[t] *: summonAll[ts]
```

Comment:
wasn't there a peephole optimisation to convert these expressions into directly calling the constructors?

Reply:
Actually, the peephole is not triggered due to the intermediate bindings.